### PR TITLE
feat(auth): fix httpsig bypass env

### PR DESCRIPTION
**Pull request by njlie**
## Changes proposed in this pull request

- 
Conditionally adds httpsig middleware depending on the setting of the env variable for bypassing httpsig validation.

## Context

This conditional used to be in the code but refactors of `app.ts` have caused it to be removed.

## Checklist

- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: packages/auth/src/app.ts [hunk 1]
```diff
@@ -195,7 +195,9 @@ export class App {
         path: '/',
         method: HttpMethod.POST
       }),
-      grantInitiationHttpsigMiddleware,
+      this.config.bypassSignatureValidation
```
Comment (wilsonianb) [inline packages/auth/src/app.ts:198]: Should the existing bypass be removed?
https://github.com/interledger/rafiki/blob/d808c85c428bfc5bc1193882905ce5fdc8fa5888/packages/auth/src/signature/middleware.ts#L25-L28

**Review by wilsonianb — APPROVED** [review]
This should work until:
- https://github.com/interledger/open-payments/issues/209

at which point OpenAPI validation will fail without the signature headers.
See:
- https://github.com/interledger/rafiki/pull/683#issuecomment-1302329483

File: packages/auth/src/app.ts (hunk 1 above)
Comment (njlie) [reply packages/auth/src/app.ts:198]: it is pretty redundant now, yeah.

File: packages/auth/src/signature/middleware.test.ts [hunk 2]
```diff
@@ -405,55 +405,6 @@ describe('Signature Service', (): void => {
...
     })
 
-    test('middleware succeeds if BYPASS_SIGNATURE_VALIDATION is true with bad signature', async (): Promise<void> => {
-      const altDeps = await initIocContainer({
-        ...Config,
-        bypassSignatureValidation: true
-      })
-
-      const altContainer = await createTestApp(altDeps)
-      appContainers.push(altContainer)
```
Comment (wilsonianb) [inline packages/auth/src/signature/middleware.test.ts:415]: You could also revert `appContainers` to `appContainer`
https://github.com/interledger/rafiki/pull/683/files#diff-9c6c5d794c9dfa6afb4129ccc2ffb701bcdecb70308c63c7e4d615e8d903859c

**Review by wilsonianb — APPROVED** [review]

**Pull request merged**